packages/account/src/Components/address-details/__tests__/address-details.spec.tsx

```diff
@@ -268,4 +274,30 @@ describe('<AddressDetails/>', () => {
         expect(screen.getByLabelText(address_town_marked)).toBeEnabled();
         expect(screen.getByLabelText(address_postcode)).toBeEnabled();
     });
+
+    it('should show validation error for postcode if adding space in the beginning', async () => {
+        renderComponent({});
+
+        const address_postcode_input: HTMLInputElement = screen.getByLabelText(address_postcode);
+        await act(async () => {
+            userEvent.type(address_postcode_input, '  2222');
+            userEvent.tab();
+        });
+
+        mockedSplitValidationResultTypes.mockReturnValue({
+            errors: {
+                address_postcode: 'Only letters, numbers, space and hyphen are allowed.',
+            },
+            warnings: {},
+        });
+
+        const next_btn = screen.getByRole('button', { name: /next/i });
+        await act(async () => {
```

Comment:
fireEvent is already wrapped to act under the hood,
if you get the warning about act, try to wrap expectations to waitFor(()=>{}),
in addition, we tend to use userEvent in deriv.app...

Reply:
Thanks @yauheni-deriv for the insights. I see we're not wrapping the side effects like state updates in `act` and the warning comes up.